pkg/pod/pod.go

```diff
@@ -336,3 +342,22 @@ func shouldOverrideWorkingDir(kubeclient kubernetes.Interface) bool {
 	}
 	return true
 }
+
+// shouldAddReadyAnnotationonPodCreate returns a bool indicating whether the
+// controller should add the `Ready` annotation when creating the Pod. The
+// default behavior is to add the Ready annotation after  Pod has been created
+// and all of its sidecars are in a Ready state.
+//
+// For further reference see https://github.com/tektoncd/pipeline/issues/2080
```

Comment:
i'm not a huge fan of referencing issues in docstrings and comments, unless it's a TODO. id rather duplicate the info here if we think it's important. Also if we link the issue from the commit message we dont need to mention it here too

Reply:
removed and added a line about why we are doing this (i.e. the injected sidecar resource)